apps/docs/src/guide/provider/pagination.md

```diff
@@ -0,0 +1,65 @@
+
+## Pagination Arguments
+
+The pagination arguments object is used to specify the range of data you want to retrieve. It includes the following properties:
+
+- `after`: A cursor pointing to a position after which you want to retrieve items.
+- `first`: The number of items to retrieve after the specified cursor. This is used in conjunction with the `after` argument.
+- `before`: A cursor pointing to a position before which you want to retrieve items.
+- `last`: The number of items to retrieve before the specified cursor. This is used in conjunction with the `before` argument.
+
+<<< @/../../docs-snippets/src/guide/provider/pagination.test.ts#pagination-1{ts:line-numbers}
+
+## Page Info
+
+The `pageInfo` object is included in the GraphQL response for requests that support cursor pagination. It provides crucial metadata about the current page of results, allowing you to understand the pagination state and determine if there are more items to fetch before or after the current set.
+
+- `endCursor`: A cursor representing the last item in the current set of results. It should be used as the `after` argument in subsequent queries to fetch the next set of items.
+- `hasNextPage`: A boolean indicating whether there are more items available after the current set.
+- `startCursor`: A cursor representing the first item in the current set of results. It should be used as the `before` argument in subsequent queries to fetch the previous set of items.
+- `hasPreviousPage`: A boolean indicating whether there are more items available before the current set.
+
+<<< @/../../docs-snippets/src/guide/provider/pagination.test.ts#pagination-2{ts:line-numbers}
+
+## Using Pagination
+
+One of the methods that supports pagination is the `getCoins` method. This method receives three parameters:
+
+- `address`: The owner's account address
+- `assetId`: The asset ID of the coins (optional)
+- `paginationArgs`: The pagination arguments (optional)
+
+### Basic Pagination
+
+Here is how you can use the `getCoins` method with pagination:
+
+<<< @/../../docs-snippets/src/guide/provider/pagination.test.ts#pagination-3{ts:line-numbers}
+
+### Navigating to the Previous Page
+
+You can also use the `paginationArgs` to navigate to the previous page of results:
+
+<<< @/../../docs-snippets/src/guide/provider/pagination.test.ts#pagination-4{ts:line-numbers}
+
+## Valid Combinations
+
+- Forward Pagination:
+
+  Use `after` with `first` to retrieve items following a cursor.
+
+<<< @/../../docs-snippets/src/guide/provider/pagination.test.ts#pagination-5{ts}
+
+- Backward Pagination:
+
+  Use `before` with `last` to retrieve items preceding a cursor.
+
+<<< @/../../docs-snippets/src/guide/provider/pagination.test.ts#pagination-6{ts}
+
+## Default Behavior
+
+If neither `assetId` nor `paginationArgs` are provided, the `getCoins` method will default to the base asset ID and return the first 100 items:
```

Comment:
Where did this `100` come from?

Do we protect against huge numbers? (i.e. a user passes `1000000000`)

Reply:
cc @xgreenx

Reply:
https://github.com/FuelLabs/fuel-core/pull/1676

Reply:
@arboleya I've defined a max value [here](https://github.com/FuelLabs/fuels-ts/pull/2408/commits/0f705a413fc3b17bc8b4cf2268fa244c346036b1)